stable/external-dns/templates/secret.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if or .Values.aws.assumeRoleArn (and .Values.aws.credentials.secretKey .Values.aws.credentials.accessKey) .Values.cloudflare.apiKey .Values.digitalocean.apiToken .Values.google.serviceAccountKey (and .Values.infoblox.wapiUsername .Values.infoblox.wapiPassword) .Values.rfc2136.tsigSecret .Values.pdns.apiKey }}
+{{- if or .Values.aws.assumeRoleArn (and .Values.aws.credentials.secretKey .Values.aws.credentials.accessKey) .Values.cloudflare.apiToken .Values.cloudflare.apiKey .Values.digitalocean.apiToken .Values.google.serviceAccountKey (and .Values.infoblox.wapiUsername .Values.infoblox.wapiPassword) .Values.rfc2136.tsigSecret .Values.pdns.apiKey }}
```

Comment:
If we are going to need one of them mandatory change:
```diff
- .Values.cloudflare.apiToken .Values.cloudflare.apiKey
+ ( eq .Values.provider "cloudflare" )
```

Reply:
But doesn't it meant we're not checking the `apiToken` & `apiKey` values? so when user empties both of that values, this condition can not handle it, right?

Reply:
But that is checked after in the if-else conditions.

Reply:
Using the required as I commented in the `apiKey` on line 24, it must be provided if an `apiToken` is not provided, otherwise the installation will fail.

Reply:
ok, fixing it

Reply:
btw @miguelaeh , how about:
```
(or .Values.cloudflare.apiToken .Values.cloudflare.apiKey)
```
just for double check 😁

Reply:
I think it could be changed too, as in case the apitoken is not provided will mount the secret for apikey and if the apikey is not provided it will fails because of the required in the secret.